language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -211,64 +215,81 @@ typeOf (Constant _ (BuiltinName _ n)) = do
 typeOf (Constant _ (BuiltinInt _ n _))           = pure (integerType n)
 typeOf (Constant _ (BuiltinBS _ n _))            = pure (bsType n)
 typeOf (Constant _ (BuiltinSize _ n))            = pure (sizeType n)
-typeOf (Apply x fun arg) = do
```

Comment:
Could you not rename all the arguments back again? That was deliberate because it makes it a lot clearer. `ty'''` is gibberish.

Reply:
Yes please.